CMakeLists.txt

```diff
@@ -224,6 +224,21 @@ else()
   unset(ENABLE_KLEE_DEBUG) # for config.h
 endif()
 
+# Warn about mixing build types.
+# This likely a bad idea because some of LLVM's header files use the NDEBUG
+# macro which can change things like data layout.
+if (LLVM_ENABLE_ASSERTIONS AND (NOT ENABLE_KLEE_ASSERTS))
+  message(WARNING
+    "LLVM was built with assertions but KLEE will be built without them.\n"
+    "This might lead to unexpected behaviour."
```

Comment:
As explained in the near by text, LLVM header files use `NDEBUG` (e.g. `include/llvm/Transforms/Scalar/LoopPassManager.h` in upstream LLVM). This can be used to change the layout of some classes (e.g. extra debugging fields). This means that in LLVM's compiled code it thinks a class is some size `X` but in KLEE's code we will think that the class is a different size `Y`. This could lead to all sort of problems, e.g.

* Allocation and deallocation of these classes might work incorrectly
* Inline methods from the header files may corrupt data.